builtin-features/operations.inc

```diff
@@ -25,10 +25,39 @@ packToken Assign(const packToken& left, const packToken& right, evaluationData*
       // It is not possible to assign directly to
       // the global scope. It would be easy for the user
       // to do it by accident, thus:
-      if (!map || *map == TokenMap::default_global()) {
+      if (!map || *map == TokenMap::default_global()) 
+      {
         data->scope[var_name] = right;
-      } else {
-        (*map)[var_name] = right;
+      }
+      else 
+      {
+          packToken* assignee = map->find(var_name);
+
+          if (assignee)
+          {
+              switch ((*assignee)->type)
+              {
+              case REALREF:
+                 static_cast<Token<std::reference_wrapper<double>>*>(assignee->token())->val.get() = right.asDouble();
+                 break;
+
+              case INTREF:
+                  static_cast<Token<std::reference_wrapper<int64_t>>*>(assignee->token())->val.get() = right.asInt();
+                  break;
+
+              case BOOLREF:
+                  static_cast<Token<std::reference_wrapper<uint8_t>>*>(assignee->token())->val.get() = right.asBool();
+                  break;
+
+              default:
+                  (*map)[var_name] = right;
+                  break;
+              }
+          }
+          else
```

Comment:
If you remove the `if (assignee)` above you may remove this else as well, the default case on the switch will account for this case.